### PR TITLE
Device tree naming

### PR DESCRIPTION
**Pull request by drewmoseley**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description
Fix broken devicetrees on Freescale bsp systems.

### CODE REVIEW

**Comment by drewmoseley** [comment]
@lluiscampos I also pushed a commit here that fixes a few more conditional variable settings.

**Review by kacf — APPROVED** [review]

**Comment by drewmoseley** [comment]
@lluiscampos any idea on the test failures here?
`time="2020-10-02T06:59:25Z" level=warning msg="Running modprobe nf_nat failed with message: `ip: can't find device 'nf_nat'\nnf_nat                 40960  3 xt_nat,xt_MASQUERADE,iptable_nat\nmodprobe: can't change directory to '/lib/modules': No such file or directory`, error: exit status 1"`

**Comment by lluiscampos** [comment]
> @lluiscampos any idea on the test failures here?
> 
> ```
> time="2020-10-02T06:59:25Z" level=warning msg="Running modprobe nf_nat failed with message: `ip: can't find device 'nf_nat'\nnf_nat                 40960  3 xt_nat,xt_MASQUERADE,iptable_nat\nmodprobe: can't change directory to '/lib/modules': No such file or directory`, error: exit status 1"
> ```

All that warnings are from the Docker in Docker setup in CI to run the tests. They have been there since day one and are not related to this PR.

The test failure in the PR is on `test_mtls_enterprise_hsm`, as the client does not report that it is loading the key from the HSM. So the code error has to be around the modified recipes. I'll look a bit more at it.

File: meta-mender-qemu/recipes-support/gnutls/gnutls_%.bbappend [hunk 1]
```diff
@@ -1,3 +1,3 @@
 
-PACKAGECONFIG_append = " p11-kit"
+PACKAGECONFIG_append_mender-client-install = " p11-kit"
```
Comment (lluiscampos) [inline meta-mender-qemu/recipes-support/gnutls/gnutls_%.bbappend:2]: The problem with the integration test is that `p11tool` is not found in the image; in other words, `p11-kit` is not installed.

I fixed it locally with the following modification:
```suggestion
ACKAGECONFIG_append = "${@bb.utils.contains('DISTRO_FEATURES', 'mender-client-install', ' p11-kit', '', d)}"
```

I don't have a good reason as per why this change fixes the issue, I followed a bit the logic that I have seen in other places where we append something to `PACKAGECONFIG`.

@kacf @drewmoseley Does this make sense to you?

Comment (kacf) [reply meta-mender-qemu/recipes-support/gnutls/gnutls_%.bbappend:2]: Yes, it does!

Comment (lluiscampos) [reply meta-mender-qemu/recipes-support/gnutls/gnutls_%.bbappend:2]: I'll submit a new PR with this fix and conflicts resolution from master.

**Comment by lluiscampos** [comment]
Fixed and re-submitted at #1109

**Pull request closed without merging**